rust/worker/src/system/types.rs

```diff
@@ -127,10 +129,29 @@ impl<C: Component> ComponentSender<C> {
         M: Send + Debug + 'static,
     {
         self.sender
-            .send(wrap(message, tracing_context))
+            .send(WrappedMessage::new(message, None, tracing_context))
             .await
             .map_err(|_| ChannelError::SendError)
     }
+
+    pub(super) async fn wrap_and_request<M>(
+        &self,
+        message: M,
+        tracing_context: Option<tracing::Span>,
+    ) -> Result<C::Result, ChannelRequestError>
```

Comment:
One ergo question is what if we forced the Result GAT to be a Result<> ? That way this isn't `Result<Result<user_result, user_error>, ChannelRequestError>` if the user returns a Result<>

Reply:
yes, that makes sense--we can push down the ChannelRequestError wrapper into the Err then instead and avoid a wrap layer for the happy path

Reply:
after spending a while looking at this, I think it might be achievable but the types get really messy
going to punt this for now as I think we may be able to achieve this more simply for task messages, which covers a large % of sent messages
if not I'll revist